submissions/yhorodechna/side-menu-dom/js/script.js

```diff
@@ -0,0 +1,107 @@
+let _navEl;
+const getNavElement = () => {
+    if (!_navEl) {
+        _navEl = document.querySelector('#nav');
+    };
+    return _navEl;
+};
+let ACTIVE_EL_ID;
+
+function findHeaderLinkOrNavLink(clickedElement) {
+    let res;
+    let current = clickedElement;
+    while (!res && current) {
+        const { className } = current;
+        if (className) {
+            if (className.includes('nav__link')) {
+                res = {
+                    navLink: current
+                }
+                continue;
+            }
+            else if (className.includes('header__link')) {
+                res = {
+                    headerLink: current
+                }
+                continue;
+            }
+        }
+        current = current.parentNode;
+    }
+    return res;
+};
+
+function handleContainerClick({ target }) {
+    const { navLink, headerLink } = findHeaderLinkOrNavLink(target);
+
+    if (navLink) {
+        if (ACTIVE_EL_ID !== navLink.id) {
+            if (ACTIVE_EL_ID) {
+                document.querySelector('#' + ACTIVE_EL_ID).classList.remove('active');
+            };
+            ACTIVE_EL_ID = navLink.id;
+            navLink.classList.add('active');
+            showArticle(navLink);
+        };
+        getNavElement().classList.remove('nav__full');
+    } else if (headerLink) {
+        getNavElement().classList.add('nav__full');
+    };
+};
+
+function createNavList({ items, parentEl, defaultId }) {
```

Comment:
is there a reason to pass parentEl as an argument? it will stay always the same

Reply:
the same is for defaultId

Reply:
I do this for code reuse. So that js is not tied to specific html.